tests/test_parsing.py

```diff
@@ -139,6 +139,23 @@ def test_url_conversion_priv_subst_method(self):
                 self.assertEqual(key, "9e1fcd2b11435552cf236be93c7ff73058870995")
                 self.assertEqual(substituted_method_id, 27600)
 
+    def test_url_conversion_errors(self):
+        test_cases = [
+            ("complib.org/method/12345", "Not a composition."),
+            ("", "Composition ID '' is not a number."),
+            ("complib.org", "URL needs more path segments."),
+            ("complib.org/composition", "URL needs more path segments."),
+        ]
+        for input_url, exp_err_str in test_cases:
+            with self.subTest(input=input_url, exp_err_str=exp_err_str):
+                try:
+                    parse_arg(input_url)
+                except InvalidComplibURLError as e:
+                    self.assertEqual(e._error, exp_err_str)
+                    continue
+                # Make this test fail if the error was not thrown
+                raise Exception()
```

Comment:
This looks fine, the way other tests have done it is
``` python
with self.assertRaisesRegex(InvalidComplibURLError, exp_err_str):
    parse_arg(input_url)
```

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex

Reply:
Bonus test case
`https://complib.org/composition/36532?substitutedmethodid=`

Reply:
Awesome! Both changes done now.